src/win/winpty.cc

```diff
@@ -215,23 +219,25 @@ static NAN_METHOD(PtyStartProcess) {
   winpty_error_free(error_ptr);
 
   // Set return values
-  v8::Local<v8::Object> marshal = Nan::New<v8::Object>();
-  Nan::Set(marshal, Nan::New<v8::String>("innerPid").ToLocalChecked(), Nan::New<v8::Number>((int)GetProcessId(handle)));
-  Nan::Set(marshal, Nan::New<v8::String>("innerPidHandle").ToLocalChecked(), Nan::New<v8::Number>((int)handle));
-  Nan::Set(marshal, Nan::New<v8::String>("pid").ToLocalChecked(), Nan::New<v8::Number>((int)winpty_agent_process(pc)));
-  Nan::Set(marshal, Nan::New<v8::String>("pty").ToLocalChecked(), Nan::New<v8::Number>(InterlockedIncrement(&ptyCounter)));
-  Nan::Set(marshal, Nan::New<v8::String>("fd").ToLocalChecked(), Nan::New<v8::Number>(-1));
-  {
-    LPCWSTR coninPipeName = winpty_conin_name(pc);
-    std::wstring coninPipeNameWStr(coninPipeName);
-    std::string coninPipeNameStr(coninPipeNameWStr.begin(), coninPipeNameWStr.end());
-    Nan::Set(marshal, Nan::New<v8::String>("conin").ToLocalChecked(), Nan::New<v8::String>(coninPipeNameStr).ToLocalChecked());
-    LPCWSTR conoutPipeName = winpty_conout_name(pc);
-    std::wstring conoutPipeNameWStr(conoutPipeName);
-    std::string conoutPipeNameStr(conoutPipeNameWStr.begin(), conoutPipeNameWStr.end());
-    Nan::Set(marshal, Nan::New<v8::String>("conout").ToLocalChecked(), Nan::New<v8::String>(conoutPipeNameStr).ToLocalChecked());
+ {
+    // Use a new block to avoid
+    // "warning C4533: initialization of 'marshal' is skipped by 'goto cleanup'"
+    v8::Local<v8::Object> marshal = Nan::New<v8::Object>();
+    Nan::Set(marshal, Nan::New<v8::String>("innerPid").ToLocalChecked(), Nan::New<v8::Number>(static_cast<double>(GetProcessId(handle))));
+    Nan::Set(marshal, Nan::New<v8::String>("innerPidHandle").ToLocalChecked(), Nan::New<v8::Number>(static_cast<double>(reinterpret_cast<uint64_t>(handle))));
```

Comment:
```suggestion
    Nan::Set(marshal, Nan::New<v8::String>("innerPidHandle").ToLocalChecked(), Nan::New<v8::Number>(static_cast<uint32_t>(handle)));
```

Reply:
Actually looking at the handle usages expose to JS, ex: `innerPidHandle`, `handle` etc they are only used for internal implementation. I would suggest to remove them from being exposed to JS and operate of pid values instead, on the native side extract the handle when needed using `OpenProcess`.

Reply:
Yes but those are internal implemenation, I am in favor of addressing them as part of this PR. I am not a fan of these double casts.

Reply:
@rzhao271 I misunderstood your question yesterday as I couldn't look at the code at the time. These seems to only be API on the binary, it's only what's in node-pty.d.ts that impacts semver now that v1.0.0 is out.